frontend/catalyst/compilation_pipelines.py

```diff
@@ -837,6 +851,55 @@ def qjit(
             elements of this list are named sequences of MLIR passes to be executed. A ``None``
             value (the default) results in the execution of the default pipeline. This option is
             considered to be used by advanced users for low-level debugging purposes.
+        abstracted_axes (Optional(Union[
+                Sequence[Sequence[str]],
+                Dict[int, str],
+                Sequence[Dict[int, str]],
+            ])): This is an experimental feature.
+            A way to specify dynamic tensor shapes.
```

Comment:
```suggestion
        abstracted_axes (Sequence[Sequence[str]] or Dict[int, str] or Sequence[Dict[int, str]]): An experimental option to specify dynamic tensor shapes.
```

A few very minor comments here:

- For _Sphinx_ type hinting, it's fine to be a little more loose for legibility. `or` is also (from what I remember) recognized by Sphinx, and is an acceptable way of communicating union.

- The optional is not needed so much, since unless advanced python signature features are used, the fact that it is a keyword argument with a default value will indicate this to the user.

- The multi-line formatting here will not work with sphinx (see rendered screenshot)

![image](https://github.com/PennyLaneAI/catalyst/assets/2959003/977af09c-c462-4012-ac2c-2c5862311e21)